cmd/operator-sdk/run/local.go

```diff
@@ -75,7 +76,11 @@ func (c runLocalArgs) runGo() error {
 	projutil.MustInProjectRoot()
 	absProjectPath := projutil.MustGetwd()
 	projectName := filepath.Base(absProjectPath)
-	outputBinName := filepath.Join(scaffold.BuildBinDir, projectName+"-local")
+	var exeSuffix string
+	if runtime.GOOS == "windows" {
+		exeSuffix = ".exe"
+	}
+	outputBinName := filepath.Join(scaffold.BuildBinDir, projectName+"-local"+exeSuffix)
```

Comment:
This change will fix the specific problem with `run --local`, but under the hood, we use the `projutil.GoBuild()` helper.

I think a better change would be to update the underlying build helper so that it always appends `.exe` when on Windows (and the bin name doesn't already have that suffix). That way we don't accidentally miss this in other places that use use `projutil.GoBuild()`

WDYT of the following instead?
```diff
diff --git a/internal/util/projutil/exec.go b/internal/util/projutil/exec.go
index 7853da76..03ba73ca 100644
--- a/internal/util/projutil/exec.go
+++ b/internal/util/projutil/exec.go
@@ -19,6 +19,7 @@ import (
        "os"
        "os/exec"
        "path/filepath"
+       "runtime"
        "strings"
 
        log "github.com/sirupsen/logrus"
@@ -111,6 +112,9 @@ func (opts GoCmdOptions) getGeneralArgsWithCmd(cmd string) ([]string, error) {
        }
 
        if opts.BinName != "" {
+               if runtime.GOOS == "windows" && !strings.HasSuffix(opts.BinName, ".exe") {
+                       opts.BinName += ".exe"
+               }
                bargs = append(bargs, "-o", opts.BinName)
        }
        bargs = append(bargs, opts.Args...)
```

Reply:
Actually, maybe we _don't_ want to do what I just suggested :stuck_out_tongue_closed_eyes:

In the `operator-sdk build` case, we build a Go binary and then copy it into a Linux container. So perhaps the `run --local` case is the only one where we want to add `.exe`

/cc @estroz @hasbro17 Bouncing this off you to check my thinking.

Reply:
Yes, this is specific to `run --local` for the reason you give; operators *normally* run in a Linux environment, not on Windows, so I think making the change you suggest would break other things.